server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/sizetiered/SizeTieredCompactionRecoverTask.java

```diff
@@ -98,11 +101,15 @@ public void doCompaction() {
             logicalStorageGroupName,
             virtualStorageGroup,
             compactionLogFile);
-        SizeTieredCompactionLogAnalyzer logAnalyzer =
-            new SizeTieredCompactionLogAnalyzer(compactionLogFile);
-        logAnalyzer.analyze();
+        CompactionLogAnalyzer logAnalyzer = new CompactionLogAnalyzer(compactionLogFile);
+        if (compactionLogFile.getName().startsWith(tsFileManager.getStorageGroupName())) {
```

Comment:
It's better to define a method such as `isOldLogFile()` or `isLogFileVersionLowerThanZeroOneThree()` here and implement the details logic inside the method

Reply:
Resolved.